web3.js/src/transaction.ts

```diff
@@ -169,13 +211,44 @@ export class Transaction {
    */
   nonceInfo?: NonceInformation;
 
+  /**
+   * @internal
+   */
+  _message?: Message;
+
+  /**
+   * @internal
+   */
+  _json?: TransactionJSON;
+
   /**
    * Construct an empty Transaction
    */
   constructor(opts?: TransactionCtorFields) {
     opts && Object.assign(this, opts);
   }
 
+  /**
+   * @internal
+   */
+  toJSON(): TransactionJSON {
+    return {
+      recentBlockhash: this.recentBlockhash || null,
+      feePayer: this.feePayer ? this.feePayer.toJSON() : null,
+      nonceInfo: this.nonceInfo
+        ? {
+            nonce: this.nonceInfo.nonce,
+            nonceInstruction: this.nonceInfo.nonceInstruction.toJSON(),
+          }
+        : null,
+      instructions: this.instructions.map(instruction => instruction.toJSON()),
+      signatures: this.signatures.map(({publicKey, signature}) => ({
+        publicKey: publicKey.toJSON(),
+        signature: signature ? [...signature] : null,
+      })),
```

Comment:
Oh nice. I was worried that there existed the possibility of false positives because in `populate` these keys are ordered ``signature`, `publicKey` instead of `publicKey`, `signature`, but since you're destructuring and restructuring them here, there's no risk of that.